pyproject.toml

```diff
@@ -54,6 +53,7 @@ dev = [
     "pytest-mock",
     "pytest-xdist",
     "ruff",
+    "bpx>=0.4",
```

Comment:
Let's move `bpx` to it's own optional dependency. Something like:

```python
bpx = [
    "bpx>=0.4",
]
```
with the `all` dependency including it like so:

```python
all = ["pybop[plot]", "pybop[scifem]", "pybop[bpx]"]
```